components/processors/observek8sattributesprocessor/podactions.go

```diff
@@ -33,6 +35,38 @@ const (
 	OwnerKindStatefulSet           = "StatefulSet"
 )
 
+// Helper Fns
+func hasPodReadyCondition(conditions []v1.PodCondition) bool {
+	for _, condition := range conditions {
+		if condition.Type == v1.PodReady && condition.Status == v1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func isPodInitializedConditionTrue(status *v1.PodStatus) bool {
+	for _, condition := range status.Conditions {
+		if condition.Type != v1.PodInitialized {
+			continue
+		}
+
+		return condition.Status == v1.ConditionTrue
```

Comment:
Maybe change this to 
```suggestion
		if condition.Type == v1.PodInitialized && condition.Status == v1.ConditionTrue {
			return true
		}
```
to match `hasPodReadyCondition`

Reply:
Bumping this one